lib/notification.php

```diff
@@ -13,34 +13,36 @@ public static function sendMail($path){
 		if (!\OCP\User::isLoggedIn()){
 			return;
 		}
-		$email = \OCP\Config::getUserValue(\OCP\User::getUser(), 'settings', 'email', '');
+		$config = \OC::$server->getConfig();
+		$user = \OC::$server->getUserSession()->getUser();
+		$email = $user->getEMailAddress();
+		$displayName = $user->getDisplayName();
 		if (\OCP\App::isEnabled('user_ldap')){
-                        $user  = \OCP\Config::getUserValue(\OCP\User::getUser(), 'user_ldap', 'displayName', '');
-                        if (empty($user)) {
-                                $user = \OCP\User::getUser();
-                        }
-                }
-                else {
-                        $user = \OCP\User::getUser();
-                };
+			$displayName = $config->getUserValue($user->getUID(), 'user_ldap', 'displayName', '');
+		}
+		if (empty($displayName)) {
+			$displayName = $user->getUID();
```

Comment:
`return null` in `User_LDAP::getDisplayName` https://github.com/owncloud/core/blob/master/apps/user_ldap/lib/User_LDAP.php#L386 thinks differently